src/meta/test/TestTableSpec.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020-present
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fmt/format.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include "meta/ClusterInfo.h"
+
+namespace nebula {
+namespace meta {
+namespace test {
+
+/**
+ * Test extract pattern marco from table spec configuration
+ */
+TEST(MetaTest, TestExtractPatternMacro) {
+  EXPECT_EQ(nebula::meta::extractPatternMacro("date"), nebula::meta::PatternMacro::DATE);
+
+  EXPECT_EQ(nebula::meta::extractPatternMacro("date,hour"), nebula::meta::PatternMacro::HOUR);
+
+  EXPECT_EQ(nebula::meta::extractPatternMacro("date hour"), nebula::meta::PatternMacro::HOUR);
```

Comment:
I think we should enforce macros with format in bracelets "{}", in that sense, I would expect this to be INVALID

Reply:
I did quick test using {date} and fill in test.yaml configuration ends with build error, may be we need to handle escape?

Reply:
In yaml/yml， you need input encoded url? {date} -> %7Bdate%7D